README.md

```diff
@@ -959,26 +959,6 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
```

Comment:
> This rule changes the format of code, but is not reformatted automatically.

Yes, when we did a pass to remove rules that are not automatically corrected we audited all rules to determine which could be easily autocorrected. We must have determined that this and it's [partner rule](https://github.com/airbnb/swift#namespace-using-enums) were not reasonably autocorrectable.

Tooling is getting better and maybe this is more feasible now. But actually updating all source code to organize constants like this would be fairly complex and we must have determined it was not really feasible.

Reply:
@fdiaz may have thoughts

Reply:
> But actually updating all source code to organize constants like this would be fairly complex and we must have 
determined it was not really feasible.

This seems approximately as challenging to autocorrect as the [mark-within-types](https://github.com/airbnb/swift#marks-within-types) rule, which we did remove from the Style Guide until it had autocorrect available. 

Are there differences you have in mind that would make this rule more challenging to autocorrect than [mark-within-types](https://github.com/airbnb/swift#marks-within-types)?

Reply:
Let's look internally to see if we can find any documentation on how we went about deciding which rules to remove. That being said, it seems like this line of reasoning would also indicate that we should remove [namespace-using-enums](https://github.com/airbnb/swift#namespace-using-enums). Would you agree?

Reply:
In contrast to this rule, I think [namespace-using-enums](https://github.com/airbnb/swift#namespace-using-enums) actually _is_ impossible to autocorrect. 

 - For this global `private` constant rule, it would be reasonably doable to simply move a `private static` declaration to the top-level scope. This wouldn't break any accessors in the file, except ones using `self` (which could be patched by the linter).

 - In contrast, for [namespace-using-enums](https://github.com/airbnb/swift#namespace-using-enums), the linter wouldn't know what name to use for the `enum` namespace. Additionally, since these symbols are `public`, changing their name / namespace is a source-breaking change across the entire codebase (all of our existing linter tools operate on the context of an individual file).

But this does raise a good question -- what is our bar for exceptions to this rule of requiring autocorrect? 
  - For example, [switch-never-default](https://github.com/airbnb/swift#switch-never-default) and [extensions-for-protocol-conformance](https://github.com/airbnb/swift/issues/58) would both be incredibly challenging to autocorrect (they require complete knowledge of the codebase, whereas all existing linters operate on single files at a time).
  - [switch-never-default](https://github.com/airbnb/swift#switch-never-default) is still present in our style guide, despite not being autocorrectable, while [extensions-for-protocol-conformance](https://github.com/airbnb/swift/issues/58) was removed.

Reply:
Fun fact that's tangentially related to autocorrecting [namespace-using-enums](https://github.com/airbnb/swift#namespace-using-enums):

The next version of SwiftFormat will have a rule that converts classes and structs to enums if they only contain static members (nicklockwood/SwiftFormat#737). 

This doesn't create _new_ namespaces, but it does help enforce the _Caseless enums work well as namespaces because they cannot be instantiated, which matches their intent_ part of that rule.

Reply:
Based on the discussion in https://github.com/airbnb/swift/pull/34#issuecomment-443930331, I'm leaning towards thinking that we wouldn't want to require autocorrect for a rule like this (even if autocorrect is theoretically possible). This rule seems more like a "pattern" than a way to "format" code.

Reply:
We discussed internally.

@fdiaz pointed us to [this comment](https://github.com/airbnb/swift/pull/34#issuecomment-443930331). We reminded ourselves of that, per our [guiding tenets](https://github.com/airbnb/swift#guiding-tenets), rules that _reformat code_ need to be able to be reformatted automatically (barring heavily justified exceptions). Per the examples in the linked comment, rules that _reformat_ code are those that _do not change the nature of the code_.

This is why [switch-never-default](https://github.com/airbnb/swift#switch-never-default) remains in our style guide while [extensions-for-protocol-conformance](https://github.com/airbnb/swift/issues/58) was removed. A very pure example of a rule that reformats code is one that adds `// MARK`s. [extensions-for-protocol-conformance](https://github.com/airbnb/swift/issues/58) does change code but doesn't change the meaning, so it falls at the extreme end of what we would consider reformatting. I can't think of how moving methods to an extension could cause a file to no longer compile.

Reply:
Applying these learnings ^ to this example, we can see that this rule does not "reformat code". This is why it remains in the style guide and is **not** in violation of our guiding tenets.

Reply:
I edited the first comment in this thread to indicate that we have updated our thinking, to save readers time.